src/views/Rules/ListRules/index.jsx

```diff
@@ -557,17 +710,17 @@ function ListRules(props) {
         open={dialogState.open}
         title={dialogState.title}
         destructive={dialogState.destructive}
-        body={dialogBody}
+        body={dialogMode === 'delete' ? deleteDialogBody : signoffDialogBody}
         confirmText={dialogState.confirmText}
-        onSubmit={handleDialogSubmit}
+        onSubmit={() => dialogState.handleSubmit(dialogState)}
         onError={handleDialogError}
         error={dialogState.error}
-        onComplete={handleDialogComplete}
+        onComplete={dialogState.handleComplete}
         onClose={handleDialogClose}
```

Comment:
STR:

1. Click the "Delete" button of a rule card.
2. Click cancel button in the dialog

Expected: The dialog closes smoothly
Actual: I feel like I am seeing a glimpse of the revoke signoff dialog for a bit. I see a purple button for like 200 ms.

Are you able to replicate by any chance? Apologies for not catching this earlier.

Reply:
I think I see that (although mine is briefer than ~200ms). I'm surprised though, because dialogMode isn't anything except `delete` during those STR. It continues to happen even if the dialog body is simply set to `{deleteDialogBody}` though, so I think there's something else going on here. I'll dig into it more.

Reply:
Okay, so this was happening because `deleteDialogBody` ends up `null` when `handleDialogClose` fires (because its contents depend on `dialogState.item` being set). The best fix I could find was to separate setting `open: false` from resetting the rest of `dialogState`. Let me know if you have a better way.